### PR TITLE
model resource unit tests

### PR DESCRIPTION
**Pull request by surajkota**
Description of changes:
- Unit tests for model
- sdk.go is at 72.5%. overall resource at 58.6%
- use gocmp to compare spec in addition to status. Will create a separate PR to add this to other resources

Testing:
`make test`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/resource/model/testdata/sdkapi/create/create_success.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "ModelArn": "arn:aws:sagemaker:us-west-2:123456789012:model/xgboost-model"
+   }
```
Comment (mbaijal) [inline pkg/resource/model/testdata/sdkapi/create/create_success.json:3]: nit: indentation

File: pkg/resource/model/testdata/sdkapi/describe/success_primary_container.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
```
Comment (mbaijal) [inline pkg/resource/model/testdata/sdkapi/describe/success_primary_container.json:1]: can we fix a filename template such that these have a similar name - `success_primary_container.json` and `describe_success.json`

File: pkg/resource/model/testdata/test_suite.yaml [hunk 3]
```diff
@@ -2,16 +2,173 @@ tests:
...
+      - name: "Create=Valid"
+        description: "Create a new model successfully (ARN in status)."
+        given:
+          desired_state: "v1alpha1/create/desired/success_before_create.yaml"
+          svc_api:
+            - operation: CreateModelWithContext
+              output_fixture: "sdkapi/create/create_success.json"
+        invoke: Create
+        expect:
+          latest_state: "v1alpha1/create/observed/success_after_create.yaml"
```
Comment (mbaijal) [inline pkg/resource/model/testdata/test_suite.yaml:27]: same, filenames should be in pattern with name in line #16

File: pkg/resource/model/testdata/test_suite.yaml [hunk 4]
```diff
@@ -2,16 +2,173 @@ tests:
...
+      - name: "Update=NotSupported"
+        description: "This test checks if the contrller throws error for update"
+        given:
+          desired_state: "v1alpha1/update/desired/updated_base.yaml"
+          latest_state: "v1alpha1/update/desired/latest_post_create.yaml"
+        invoke: Update
+        expect:
+          latest_state: "v1alpha1/update/observed/error_on_update.yaml"
+          error: "not implemented"
+  #     - name: "Update=StatusFailed"
```
Comment (mbaijal) [inline pkg/resource/model/testdata/test_suite.yaml:107]: Cleanup ?

**Review by mbaijal — CHANGES_REQUESTED** [review]

File: pkg/resource/model/testdata/test_suite.yaml (hunk 4 above)
Comment (ryansteakley) [reply pkg/resource/model/testdata/test_suite.yaml:107]: ^ Is this commented out by mistake?

File: pkg/resource/model/testdata/sdkapi/describe/success_primary_container.json (hunk 2 above)
Comment (surajkota) [reply pkg/resource/model/testdata/sdkapi/describe/success_primary_container.json:1]: renamed to success_

File: pkg/resource/model/testdata/test_suite.yaml (hunk 3 above)
Comment (surajkota) [reply pkg/resource/model/testdata/test_suite.yaml:27]: couldnt think of better naming but I agree, there should be a pattern to make it more readable

**Review by ryansteakley — APPROVED** [review]

**Pull request merged**